### DISCUSSION TITLE
Appium reporting resolution size for Pixel4(Android) and viewport size for iPhoneXS(apple), how to choose absolute pixel location ?

### DISCUSSION THREAD

**Discussion by vnuta9** [appium/appium#18776] [Q&A] [answered]
I have some 10 method calls in my appium(ver2.0.0-beta71) QA testing code with absolute pixel measurements, which I have to correct by translating them to screen size percentage terms. I queries using Java-Client 8.5.1 for window size=driver.manage().window().getSize() which returned

-    Pixel4 >>> Window size=1080 X 2214
-    iPhoneXS >>>Window size=375 X 812
By googling I found this at https://yesviz.com/ that 

- Google Pixel 4 resolution is 1080 x 2280 PX with ~ 444 pixel desnsity. Google Pixel 4 viewport size is 393 x 830 PX with ~ 161 actual pixel density
- iPhone XS has a 5.8-inch screen with a screen size (resolution): 1125px × 2436px, 375px × 812px viewport1, 

As I also read that resolution is physical pixel count, and viewport is software pixel count.

**But why I am getting physical pixel count for Pixel4 Android phone and viewport pixel count for iPhone ?? How to reconcile this and convert in to window size percentages ?**

**Comment by jlipps** [comment, marked as answer]
Try to look at the returned server capabilities or call get session details. It might list out the display density which you can use as a multiplier. Otherwise, what is the real issue? What are you using the pixel measurements for? If it's for creating touch actions, then you use the virtual pixels anyway. The only reason you need to convert to real pixels for iOS is if you're going to do some kind of image analysis.
